### PR TITLE
Issue #2527162: Cannot call global functions that take a method by reference (follow-up)

### PR DESCRIPTION
**Pull request by drupol**

### CODE REVIEW

**Comment by drupol** [comment]
The best of both worlds ;-)

File: src/Legacy/Drupal7.php [hunk 1]
```diff
@@ -43,15 +43,47 @@
...
+        break;
+      default:
+        // This is a necessary hack to ensure that things that have been passed by
+        // reference continue to being passed by reference.
+        // More info: See http://goo.gl/aM5fra
+        $args_reference = array();
+        foreach($args as $key => &$arg) {
+          $args_reference[$key] = &$arg;
+        }
+        call_user_func_array($method, $args_reference);
```
Comment (fabianx-ai) [inline src/Legacy/Drupal7.php]: Missing return ...

everywhere, that is why travis failed.

File: src/Legacy/Drupal7.php [hunk 2]
```diff
@@ -43,15 +43,47 @@
...
 
   public function __call($method, $args) {
-    // This is a necessary hack to ensure that things that have been passed by
-    // reference continue to being passed by reference.
-    // More info: See http://goo.gl/aM5fra
-    $args_reference = array();
-    foreach($args as $key => &$arg){
-      $args_reference[$key] = &$arg;
+    switch(count($args)) {
+      case 1:
```
Comment (fabianx-ai) [inline src/Legacy/Drupal7.php:50]: The 0 case is not handled.

return $method();

**Comment by fabianx-ai** [comment]
CNW

File: src/Legacy/Drupal7.php [hunk 3]
```diff
@@ -43,15 +43,47 @@
...
+        $method($args[0]);
+        break;
+      case 2:
+        $method($method, $args[0], $args[1]);
+        break;
+      case 3:
+        $method($args[0], $args[1], $args[2]);
+        break;
+      case 4:
+        $method($args[0], $args[1], $args[2], $args[3], $args[4]);
```
Comment (fabianx-ai) [inline src/Legacy/Drupal7.php]: BUG - Travis found, needs to be only to $args[3] here.

File: src/Legacy/Drupal7.php [hunk 4]
```diff
@@ -43,15 +43,47 @@
...
+        break;
+      case 5:
+        $method($args[0], $args[1], $args[2], $args[3], $args[4], $args[5]);
```
Comment (fabianx-ai) [inline src/Legacy/Drupal7.php]: That one is wrong, too.

Pyramid structure interrupted :)

**Pull request merged**